Signal/src/AppDelegate.m

```diff
@@ -197,9 +197,10 @@ - (UIViewController *)loadingRootViewController
         [[UIStoryboard storyboardWithName:@"Launch Screen" bundle:nil] instantiateInitialViewController];
 
     BOOL shouldShowUpgradeLabel = NO;
-    NSString *previousVersion = AppVersion.instance.lastAppVersion;
+    NSString *previousVersion = AppVersion.instance.lastCompletedLaunchAppVersion;
     // We added a number of database views in v2.13.0.
-    if ([VersionMigrations isVersion:previousVersion atLeast:@"2.0.0" andLessThan:@"2.13.0"]) {
+    if ([TSAccountManager isRegistered] &&
```

Comment:
If we just omit it, people who have previously launched, but never completed registration, then upgraded signal, then decide to complete registration, will see a short blip of the upgrading screen.

I think that's a reasonable concession given the relatively small group of people.